File: ToDo/app/src/main/java/edu/mtsu/todo/ToDoFragment.java
package edu.mtsu.todo;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.text.InputType;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class ToDoFragment extends Fragment implements View.OnClickListener {
    ToDoAdapter adapter;
    LinearLayoutManager layoutManager;
    RecyclerView recyclerView;
    FloatingActionButton addButton;
    AlertDialog.Builder builder;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_todo, container, false);

        // late initialize our variables here
        recyclerView = view.findViewById(R.id.recyclerView);
        addButton = view.findViewById(R.id.addButton);
        layoutManager = new LinearLayoutManager(getContext());
        adapter = new ToDoAdapter();
        builder = new AlertDialog.Builder(getContext());

        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);

        addButton.setOnClickListener(this);

        return view;
    }

    @Override
    public void onClick(View view) {
        builder.setTitle("Add Item");

        final EditText input = new EditText(getContext());

        input.setInputType(InputType.TYPE_CLASS_TEXT);
        builder.setView(input);

        builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                adapter.addItem(input.getText().toString());
            }
        });
        builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.cancel();
            }
        });

        builder.show();
    }
}

File: ToDo/app/src/main/java/edu/mtsu/todo/ToDoFragment.kt
package edu.mtsu.todo

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.floatingactionbutton.FloatingActionButton

class ToDoFragment: Fragment() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var adapter: ToDoAdapter
    private lateinit var fab: FloatingActionButton


    // Initialize everything here
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_todo, container, false)
        recyclerView = view.findViewById(R.id.todo_recycler)
        fab = view.findViewById(R.id.fabAdd)

        // set layout manager
        // this is what controls how the rv is laid out. In our case, linearly.
        recyclerView.layoutManager = LinearLayoutManager(context)

        // set adapter
        adapter = ToDoAdapter()
        recyclerView.adapter = adapter

        fab.setOnClickListener {
            adapter.items.add("")
            adapter.notifyDataSetChanged()
        }

        return view
    }

}